assets/src/detour.ts

```diff
@@ -0,0 +1,26 @@
+import { ShapePoint } from "./schedule"
+
+export type DirectionType =
+  | "left"
+  | "right"
+  | "sharp_left"
+  | "sharp_right"
+  | "slight_left"
+  | "slight_right"
+  | "straight"
+  | "enter_roundabout"
+  | "exit_roundabout"
+  | "u_turn"
+  | "goal"
+  | "depart"
+  | "keep_left"
+  | "keep_right"
+
+export interface DetourShape {
+  coordinates: ShapePoint[]
+  directions: {
+    instruction: string
+    name: string
```

Comment:
Nit: `name` and `type` aren't being used at all, which I think has implications for the backend as well.

Reply:
I'm considering dropping these because we're not using them and they were added in speculation.
I'm hesitent because that would mean the `Skate.OpenRouteServiceAPI.map_type/1` function would no longer be used.

I'm fairly certain we will not be using ORS's generated `instruction`(speculation), but that was the easiest to do at first. Given that, do you still think it should be dropped?

Reply:
@lemald do you have any thoughts on this?

Reply:
A few options:
 - Just remove it, and we'll recreate it if needed (hopefully that wouldn't take too much effort)
 - Remove it, and create a branch reverting the removal so that it'll just be there if/when we need it
 - Mayyyyyybe... generate our own "instruction" based on `type` and `name`, rather than using ORS's? Even if our result would be the same as ORS's, this would still have the advantage of making pretty easy to make any adjustments in the future

I'm fine with any of these... I think I have a slight preference for the third option.

Reply:
Just noting this, I found the localization file which actually generates the instructions `https://github.com/GIScience/openrouteservice/blob/ab2be549acc21455a2ad7ca925c74ffb9db39cd9/ors-engine/src/main/resources/resources/locales/ors_en-US.resources#L64`

Something good to keep in mind, this shows that for roundabout instructions, there's an extra `exit_number` value to parse out of the object.

Reply:
I opted to remove it, because we don't know the design requirements here yet.
My thinking is
- We _could_ format it on the **frontend**
	- Pro: gives us a lot of flexibility in the rendering part
		- like wrapping the instruction (`"left"`, `"right"`, `"sharp right"`) and the name in `<strong>` or `<em>`
			- ex: 
			  > **Turn right** onto _Boylston St_
	- Con: Potentially need to calculate the instruction twice, once on frontend, and once on backend, to ensure data integrity when writing to the DB
- We _could_ format it on the **backend**
	- Pro: all the data is there, less stuff to send to frontend
	- Con: but then the frontend can't fancy rendering stuff as easily,